### ISSUE TITLE
Default file and folder permissions

### ISSUE THREAD

**Issue by xchs** [contao/core-bundle#882]
I have set new default file and folder permissions, but Contao does not seem to respect these chmod settings.

To reproduce:
* add the new file and folder permissions to the `localconfig.php` configuration file:
```php
$GLOBALS['TL_CONFIG']['defaultFileChmod'] = 0664;
$GLOBALS['TL_CONFIG']['defaultFolderChmod'] = 0775;
```
* switch to the Contao back end
* <kbd>Layout</kbd> > <kbd>Templates</kbd> > Create a new template `foo` in the target folder
* <kbd>System</kbd> > <kbd>File manager</kbd> > Create a new folder `bar`
* check the file and folder permissions for `foo` and `bar`, respectively

**Comment by fritzmg** [comment]
As far as I can see `defaultFileChmod` is only used for file uploads and when saving the `localconfig.php`. It was the same in Contao 3.

`defaultFolderChmod` was only used with the "Safe Mode Hack" in Contao 3, which is not present anymore in Contao 4.

**Comment by xchs** [comment]
Thanks, However, the chmod settings should be applied for **all** writing operations that Contao might perform somewhere (imho).

**Comment by fritzmg** [comment]
But why would it be needed in the first place?

**Comment by xchs** [comment]
I have to deal with a system, where my (FTP) user has only write permissions on group level. That said, in this case it would be quite helpful if Contao sets the permissions for files and folders accordingly (i.e. 664 and 775).

**Comment by leofeyer** [comment]
> However, the chmod settings should be applied for **all** writing operations

I agree.

@contao/developers I guess the simplest way to fix this would be adding `chmod()` calls in the `Files` class. Then we only have to adjust the places where we are using the Symfony `Filesystem` class. What do you think?

**Comment by aschempp** [comment]
in my opinion this is a server misconfiguration…

**Comment by leofeyer** [comment]
As discussed in Mumble on July 13th, the problem is not that the `chmod()` calls are missing but that there are still some left-over `chmod()` calls that should have been removed together with the safe mode hack. Therefore we want to remove the left-over calls.

**Comment by xchs** [comment]
`chmod` ≠ `chown`

The Safe Mode Hack does not matter here at all! The purpose of the SMH was only to execute PHP scripts with FTP user rights.

This ticket is all about setting access permissions, no matter which user performs those operation. Why shouldn't it be possible to tell Contao which `chmod` settings to apply for the write operation?

**Comment by leofeyer** [comment]
But we have only added this feature for the safe mode hack:

https://github.com/contao/core/blob/master/system/modules/core/library/Contao/Files/Ftp.php#L113

**Comment by leofeyer** [comment]
Fixed in 039255a20c6f940a2d56d288dabda7a4b93855f1.
